pkg/manager/collectors/harvester.go

```diff
@@ -0,0 +1,81 @@
+package collectors
+
+import (
+	"path/filepath"
+
+	"github.com/Jeffail/gabs/v2"
+	"github.com/sirupsen/logrus"
+)
+
+type harvesterModule struct {
+	c    *common
+	name string
+}
+
+func NewHarvesterModule(common *common, name string) *harvesterModule {
+	return &harvesterModule{
+		c:    common,
+		name: name,
+	}
+}
+
+func (module harvesterModule) generateYAMLs() {
+	logrus.Infof("[%s] generate YAMLs, yamlsDir: %s", module.name, module.c.yamlsDir)
+
+	extraResources := getHarvesterExtraResource()
+	for namespace, resourceLists := range extraResources {
+		dir := filepath.Join(module.c.yamlsDir, "namespaced", namespace)
+		objs, err := module.c.discovery.SpecificResourcesForNamespace(module.toObj, module.name, namespace, resourceLists, module.c.errorLog)
+
+		if err != nil {
+			logrus.Error("Unable to fetch namespaced resources")
+			return
+		}
+
+		for name, obj := range objs {
+			file := filepath.Join(dir, name+".yaml")
+			logrus.Debugf("Prepare to encode to yaml file path: %s", file)
+			module.c.encodeFunc(obj, file, module.c.errorLog)
+		}
+	}
+}
+
+func (module harvesterModule) toObj(b []byte, groupVersion, kind string, resources ...string) (interface{}, error) {
+	jsonParsed, err := module.c.toObjCommon(b, groupVersion, kind)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if len(resources) > 1 {
+		logrus.Debugf("Should not have more than one resources, reousrce: %v", resources)
+	}
+	/* It's a trick for general purpose, we assume that resources should only have one element */
+	switch resources[0] {
```

Comment:
Looks a bit weird here. Why not wrap this in for loop and log the unknown resources in the default case?

Reply:
Ah...
yap, I thought you are right. Already update this for logging the unknown resource for the for-loop checking.
Thanks for the tips! Please help to review it again.